src/lib.rs

```diff
@@ -468,6 +468,20 @@ impl TrayIcon {
     pub fn window_handle(&self) -> windows_sys::Win32::Foundation::HWND {
         self.tray.borrow().hwnd()
     }
+
+    /// Get the tray icon's underlying [NSStatusItem](objc2_app_kit::NSStatusItem) **macOS only**.
+    ///
+    /// Returns `None` if the status item is not available.
+    #[cfg(target_os = "macos")]
+    pub fn ns_status_item(&self) -> Option<objc2::rc::Retained<objc2_app_kit::NSStatusItem>> {
+        self.tray.borrow().ns_status_item().cloned()
+    }
+
+    /// Get the tray icon's underlying [AppIndicator](libappindicator::AppIndicator) **Linux only**.
+    #[cfg(all(unix, not(target_os = "macos")))]
+    pub fn app_indicator(&self) -> &libappindicator::AppIndicator {
+        unsafe { &*(self.tray.borrow().app_indicator() as *const _) }
+    }
```

Comment:
This is unsafe and we should convey that in the method signature.

```suggestion
    /// Get the tray icon's underlying [AppIndicator](libappindicator::AppIndicator) **Linux only**.
    ///
    /// # Safety
    ///
    /// The returned pointer is valid as long as the `TrayIcon` is.
    #[cfg(all(unix, not(target_os = "macos")))]
    unsafe pub fn app_indicator(&self) -> *const libappindicator::AppIndicator {
        unsafe { self.tray.borrow().app_indicator() as *const _ }
    }
```